### PR TITLE
Add internal package to libnavigation-util

### PR DESCRIPTION
**Pull request by Guardiola31337**
## Description

This PR adds `internal` package to `libnavigation-util`

Splitting https://github.com/mapbox/mapbox-navigation-android/issues/2417 in different PRs

Follow up from https://github.com/mapbox/mapbox-navigation-android/pull/2811

As this is going to be a _big_ refactor the plan here is to open _small_ PRs and move stuff incrementally making the review easier. Even trying to do so this first PR included changes in 64 files (mostly moving files around)

- [x] I have added any issue links
- [x] I have added all related labels (`bug`, `feature`, `new API(s)`, `SEMVER`, etc.)
- [x] I have added the appropriate milestone and project boards

### Goal

The goal here is to add `internal` package to `1.0` modules and the APIs designed only for local usage

### Implementation

- Move classes around to `internal` package
- Add `internal` modifier
- Fix package across codebase

## Testing

- [x] I have tested locally (including `SNAPSHOT` upstream dependencies if needed) through testapp/demo app and run all activities to avoid regressions
- [x] I have tested via a test drive, or a simulation/mock location app
- [x] New and existing unit tests pass locally with my changes

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have updated the `CHANGELOG` including this PR

### CODE REVIEW

File: libtrip-notification/src/main/java/com/mapbox/navigation/trip/notification/utils/time/span/SpanItem.kt [hunk 1]
```diff
@@ -0,0 +1,6 @@
+package com.mapbox.navigation.trip.notification.utils.time.span
+
+internal interface SpanItem {
+
```
Comment (kmadsen) [inline libtrip-notification/src/main/java/com/mapbox/navigation/trip/notification/utils/time/span/SpanItem.kt:4]: want this space?

File: libnavigation-core/src/main/java/com/mapbox/navigation/core/MapboxNavigation.kt [hunk 2]
```diff
@@ -554,14 +554,18 @@ constructor(
                 arrayOf(
                     MapboxNativeNavigator::class.java to MapboxNativeNavigatorImpl,
                     MapboxOnboardRouterConfig::class.java to (navigationOptions.onboardRouterConfig
-                        ?: throw RuntimeException(MAPBOX_NAVIGATION_TOKEN_EXCEPTION_ONBOARD_ROUTER))
+                        ?: throw RuntimeException(MAPBOX_NAVIGATION_TOKEN_EXCEPTION_ONBOARD_ROUTER)),
+                    Logger::class.java to MapboxModuleProvider.createModule(
+                        MapboxModuleType.CommonLogger,
+                        ::paramsProvider
```
Comment (Guardiola31337) [inline libnavigation-core/src/main/java/com/mapbox/navigation/core/MapboxNavigation.kt:560]: Regression from https://github.com/mapbox/mapbox-navigation-android/pull/2774 cc @Lebedinsky @LukasPaczos

File: libnavigation-core/src/main/java/com/mapbox/navigation/core/MapboxNavigation.kt [hunk 3]
```diff
@@ -554,14 +554,18 @@ constructor(
...
+                    )
                 )
             }
             MapboxModuleType.NavigationTripNotification -> arrayOf(
                 Context::class.java to context.applicationContext,
                 NavigationOptions::class.java to navigationOptions
             )
-            MapboxModuleType.CommonLogger -> arrayOf()
+            MapboxModuleType.CommonLogger -> arrayOf(Logger::class.java to logger)
```
Comment (Guardiola31337) [inline libnavigation-core/src/main/java/com/mapbox/navigation/core/MapboxNavigation.kt:568]: Regression from https://github.com/mapbox/mapbox-navigation-android/pull/2774 cc @Lebedinsky @LukasPaczos

File: libnavigation-core/src/main/java/com/mapbox/navigation/core/MapboxNavigation.kt [hunk 4]
```diff
@@ -554,14 +554,18 @@ constructor(
                 arrayOf(
                     MapboxNativeNavigator::class.java to MapboxNativeNavigatorImpl,
                     MapboxOnboardRouterConfig::class.java to (navigationOptions.onboardRouterConfig
-                        ?: throw RuntimeException(MAPBOX_NAVIGATION_TOKEN_EXCEPTION_ONBOARD_ROUTER))
+                        ?: throw RuntimeException(MAPBOX_NAVIGATION_TOKEN_EXCEPTION_ONBOARD_ROUTER)),
+                    Logger::class.java to MapboxModuleProvider.createModule(
```
Comment (LukasPaczos) [inline libnavigation-core/src/main/java/com/mapbox/navigation/core/MapboxNavigation.kt:558]: this could be `Logger::class.java to logger`

**Pull request merged**